### PR TITLE
Only show necessary fields for ordering service

### PR DESCRIPTION
**Pull request by baixinsui**
fixed https://github.com/eclipse-xpanse/xpanse/issues/940
based on https://github.com/eclipse-xpanse/xpanse/pull/944
chrome_MLZpPafA2T
chrome_vZpn2KDCDq

### CODE REVIEW

**Comment by swaroopar** [comment]
we can merge after the main API changes PR is merged.. in case there are further changes.

**Comment by baixinsui** [comment]
> we can merge after the main API changes PR is merged.. in case there are further changes.

I merged the PR update APIs

File: src/xpanse-api/generated/models/UserOrderableServiceVo.ts [hunk 1]
```diff
@@ -8,67 +8,60 @@
...
+     * The variables for the deployment, which will be passed to the deployer.
      */
-    lastModifiedTime: string;
+    variables: Array<DeployVariable>;
     /**
-     * State of service.
+     * The flavors of the orderable service.
      */
-    serviceRegistrationState: ServiceTemplateVo.serviceRegistrationState;
+    flavors: Array<Flavor>;
```
Comment (swaroopar) [inline src/xpanse-api/generated/models/UserOrderableServiceVo.ts:57]: @baixinsui should we generate the client again?

File: src/components/content/common/catalog/userServiceProps.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { UserOrderableServiceVo } from '../../../../xpanse-api/generated';
```
Comment (swaroopar) [inline src/components/content/common/catalog/userServiceProps.ts:1]: this class can be moved under order/services/ package

File: src/components/content/order/query/userOrderableServicesQuery.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+import { ServiceCatalogService, ServiceVo } from '../../../../xpanse-api/generated';
+
+export default function userOrderableServicesQuery(category: ServiceVo.category, serviceName: string | undefined) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (swaroopar) [inline src/components/content/order/query/userOrderableServicesQuery.ts:5]: we must check and handle this exception

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: src/xpanse-api/generated/models/UserOrderableServiceVo.ts (hunk 1 above)
Comment (baixinsui) [reply src/xpanse-api/generated/models/UserOrderableServiceVo.ts:57]: The field 'price' in  the class Flavor is necessary, so there is no need to regenerate it

File: src/components/content/common/catalog/userServiceProps.ts (hunk 2 above)
Comment (baixinsui) [reply src/components/content/common/catalog/userServiceProps.ts:1]: Ok.

File: src/components/content/order/query/userOrderableServicesQuery.ts (hunk 3 above)
Comment (baixinsui) [reply src/components/content/order/query/userOrderableServicesQuery.ts:5]: image
Rename the  function name to 'UserOrderableServicesQuery' start with upper letter, the error is fixed.
image

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**